### PR TITLE
XamarinSupport

### PR DESCRIPTION
**Pull request by jmmortega**
I need support for Xamarin in MvvmCross projects. 

I change the serialization, for Json.Net serialization and the post request for Mixpanel's API, because in testing I receive a exception because MixPanel rejects. But I add the data in the Uri Post Request and works fine.

### CODE REVIEW

**Comment by chrisnicola** [comment]
Does Mono not have built in JSON serialization?

**Comment by jmmortega** [comment]
I use the library in a PCL library and this don't support JavaScriptSerializer, also the Json.Net perfomance is better. http://james.newtonking.com/images/jsonperformance.png

**Comment by chrisnicola** [comment]
@jmmortega it is but it means adding a dependency just for doing some fairly limited Json serialization. I won't merge this as is though obviously anyone who wants to use Json.net is free to use the fork. 

However I do realize most people do use Json.net in their projects anyways (I do myself) so I'd be very welcoming of a refactor that allows us to use a provider model so that anyone can replace the default JSON implementation with whatever they like/use. Then we can include a provider for Json.net in the project. Does this make sense?

Basically this is similar to how some libraries let you change the dependency injection library you want to use.
